web/src/click_handlers.js

```diff
@@ -388,6 +407,46 @@ export function initialize() {
         $row.find(".markdown_preview").show();
     });
 
+    $("body").on("mouseenter", ".compose_control_button", () => {
+        const buttons = document.querySelectorAll(`
+            .compose_control_button[data-tooltip-template-id],
+            .compose_control_button[data-tippy-content],
+            .compose_control_button_container
+        `);
+
+        const tippyInstances = [...buttons].map((button) => {
+            const tippyContent = button.dataset.tippyContent;
+            const tooltipTemplateId = button.dataset.tooltipTemplateId;
+
+            if (tippyContent) {
+                return tippy(button, {
+                    content: tippyContent,
+                });
+            } else if (tooltipTemplateId) {
+                const template = document.querySelector(`#${tooltipTemplateId}`);
+                const templateContent = template ? render_html_template(template.innerHTML) : "";
+
+                return tippy(button, {
+                    content: templateContent,
+                });
+            }
+            return null;
+        });
+
+        const validTippyInstances = tippyInstances.filter((instance) => instance !== null);
+
+        createSingleton(validTippyInstances, {
+            delay: 500,
+            onTrigger(instance, event) {
+                if (event.currentTarget.classList.contains("disabled-on-hover")) {
+                    instance.setProps({delay: 0});
+                } else {
+                    instance.setProps({delay: 500});
+                }
+            },
```

Comment:
Also, it's important to know that Singletons are also instances, and thus we are also creating multiple singletons over here, which we want to also avoid. On top of that, we should also take into account the [following](https://atomiks.github.io/tippyjs/v6/addons/#destroy):

> When you call singleton.destroy(), the tippyInstances you passed as an argument will not be destroyed also. They are separate instances that can be reused again elsewhere. You should also destroy the tippy instances upon cleanup.